### PR TITLE
core: generate mermaid syntax and render visual graph

### PR DESCRIPTION
**Pull request by angeligareta**
- **Description:** Add functionality to generate Mermaid syntax and render flowcharts from graph data. This includes support for custom node colors and edge curve styles, as well as the ability to export the generated graphs to PNG images using either the Mermaid.INK API or Pyppeteer for local rendering.
- **Dependencies:** Optional dependencies are `pyppeteer` if rendering wants to be done using Pypeteer and Javascript code.

### CODE REVIEW

File: libs/core/langchain_core/runnables/graph_mermaid.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        _render_mermaid_using_api(mermaid_syntax,output_file)
+    else:
+        raise ValueError(f"Invalid draw method: {draw_method}. Supported draw methods are: {', '.join([m.value for m in MermaidDrawMethod])}")
+
+async def _render_mermaid_using_pyppeteer(mermaid_syntax: str, output_file: str):   
+    """Renders the Mermaid graph using Pyppeteer and saves it as a PNG image. It requires Pyppeteer to be installed. """ 
+    try:
+        from pyppeteer import launch
+    except ImportError:
+        raise ImportError("Install Pyppeteer to use the Pyppeteer method: `pip install pyppeteer`.")
```
Comment (nfcampos) [inline libs/core/langchain_core/runnables/graph_mermaid.py:158]:
```suggestion
    except ImportError as e:
        raise ImportError("Install Pyppeteer to use the Pyppeteer method: `pip install pyppeteer`.") from e
```

File: libs/core/langchain_core/runnables/graph_mermaid.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    })
+    await page.screenshot({'path': output_file, 'fullPage': False})
+    await browser.close()
+
+def _render_mermaid_using_api(mermaid_syntax: str, output_file: str):
+    """Renders the Mermaid graph using the Mermaid.INK API and saves it as a PNG image."""
+    import base64
+    try:
+        import requests
+    except ImportError:
```
Comment (nfcampos) [inline libs/core/langchain_core/runnables/graph_mermaid.py:200]: same as above

File: libs/core/langchain_core/runnables/graph_mermaid.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            document.body.style.background = 'white'; // Set the page background to white
+        }''', svg_code['svg'])
+
+    # Take a screenshot
+    dimensions = await page.evaluate('''() => {
+            const svgElement = document.querySelector('svg');
+            const rect = svgElement.getBoundingClientRect();
+            return { width: rect.width, height: rect.height };
+        }''')
+    padding = 10  # Reduce padding to 10px or to a value that works for your case
```
Comment (nfcampos) [inline libs/core/langchain_core/runnables/graph_mermaid.py:187]: worth making configurable>?

Comment (angeligareta) [reply libs/core/langchain_core/runnables/graph_mermaid.py]: Good point, added both that and the background color

**Review by nfcampos — APPROVED** [review]

**Pull request merged**